lrpar/src/lib/ctbuilder.rs

```diff
@@ -733,17 +734,23 @@ where
         // Record the time that this version of lrpar was built. If the source code changes and
         // rustc forces a recompile, this will change this value, causing anything which depends on
         // this build of lrpar to be recompiled too.
-        writeln!(cache, "   Build time: {:?}", env!("VERGEN_BUILD_TIMESTAMP")).ok();
-
-        writeln!(cache, "   Grammar path: {:?}", self.grammar_path).ok();
-        writeln!(cache, "   Mod name: {:?}", self.mod_name).ok();
-        writeln!(cache, "   Recoverer: {:?}", self.recoverer).ok();
-        writeln!(cache, "   YaccKind: {:?}", self.yacckind).ok();
-        writeln!(cache, "   Visibility: {:?}", self.visibility.cow_str()).ok();
+        let build_time = env!("VERGEN_BUILD_TIMESTAMP");
+        let grammar_path = self.grammar_path.as_ref().unwrap().to_string_lossy();
+        let mod_name = self.mod_name;
+        let recoverer = self.recoverer;
+        let yacckind = self.yacckind;
+        let visibility = self.visibility.cow_str();
+        let error_on_conflicts = self.error_on_conflicts;
+        writeln!(cache, "   Build time: {}", quote!(#build_time)).ok();
+        writeln!(cache, "   Grammar path: {}", quote!(#grammar_path)).ok();
+        writeln!(cache, "   Mod name: {}", quote!(#mod_name)).ok();
+        writeln!(cache, "   Recoverer: {}", quote!(#recoverer)).ok();
+        writeln!(cache, "   YaccKind: {}", quote!(#yacckind)).ok();
+        writeln!(cache, "   Visibility: {}", quote!(#visibility)).ok();
```

Comment:
All these cache lines will now include fully qualified symbols where before they just used e.g.
```RecoveryKind::CPCTPlus` If I understand correctly the only effect changing it will cause is there will be an extra complete rebuild since the cache changed.

Reply:
That's fine: a one-off rebuild isn't going to trouble anyone.